### PR TITLE
vmagent: use `-enableMultitenantHandlers` command-line flag for vmage…

### PR DESCRIPTION
**Pull request by Haleygo**
…nt remote write multiTenant mode, previous flag `remoteWrite.multitenantURL` is already deprecated.

### CODE REVIEW

**Review by AndrewChubatiuk — DISMISSED** [review]

File: internal/controller/operator/factory/vmagent/vmagent.go [hunk 1]
```diff
@@ -1106,9 +1106,12 @@ func buildRemoteWrites(cr *vmv1beta1.VMAgent, ssCache *scrapesSecretsCache) []st
 	remoteTargets := cr.Spec.RemoteWrite
 
 	url := remoteFlag{flagSetting: "-remoteWrite.url=", isNotNull: true}
+	var multiTenantModeSuffix string
 	if cr.Spec.RemoteWriteSettings != nil && cr.Spec.RemoteWriteSettings.UseMultiTenantMode {
```
Comment (f41gh7) [inline internal/controller/operator/factory/vmagent/vmagent.go]: I'd suggest do not modify remoteWrite url path. It brings more harm then good.

 vmagent may accept data with multitenant handlers and forward it further to remote storage without such support. For instance, chain of vmagents or vmagent and prometheus.

 Just mention breaking change at change log and provide solution for configuration migration.

Comment (f41gh7) [reply internal/controller/operator/factory/vmagent/vmagent.go]: My motivation is:
- useMultiTenantMode  is not a popular option and it's fine to break compatibility for it.
- it makes behavior obvious for people just started to use this option.

Comment (Haleygo) [reply internal/controller/operator/factory/vmagent/vmagent.go]: I like the chain vmagent idea! Updated, please check.

**Review by f41gh7 — APPROVED** [review]
LGTM

**Comment by f41gh7** [comment]
Thanks for contribution!

**Pull request merged**